### ISSUE TITLE
Choosing any addin from the Addins menu requires the installation of {miniUI} even if the package isn't actually needed

### ISSUE THREAD

**Issue by daattali** [rstudio/rstudio#12078]
I wrote a simple RStudio addin and wanted to open it from the Addins menu. RStudio is forcing me to install {miniUI}, claiming that it must be intalled in order to run the addin. My addin does not use {miniUI} at all.

I know that when addins first got introcued, there was a push to use {miniUI}, but since it never got too much adoption, are we really still required to install it?

RStudio Server 1.4.1717

**Comment by kevinushey** [comment]
Throwing out some options:

- We should introduce a `Depends:` field into the `inst/rstudio/addins.dcf` file, and use that to infer what other packages are required by a particular addin.

- We could infer a dependency on `miniUI` only for interactive addins (under the old assumption that an interactive addin would use `miniUI`)

- We could automatically try to infer an addin's dependencies using e.g. `renv::dependencies()` on the function body registered for the addin.

**Comment by daattali** [comment]
Regarding point 2: many interactive addins (including the one that prompted this issue) do not use {miniUI}

I'd liek to offer a 4th option: simply run the function that the addin was calling, without trying to infer anything. Presumably the addin is in the *Addins* because its package is installed. Usually when a package is installed, its dependencies are already installed with it. I can successfully call the function that the addin is calling from the RStudio command line. Invoking the function call from the command line and clicking the button in the *Addins* menu should be the same thing. There's no safety check when calling the function from the command line, I would just get an error if a dependency isn't installed, so I fail to see why additional safety checks are done when using the *Addins* menu.

**Comment by gadenbuie** [comment]
I agree with @daattali. I think at this point in time, forcing installation of `miniUI` is overly prescriptive. Many addins do not require `miniUI` and this check has likely inspired changes in R tools to avoid the dialog, such as `devtools` depending on `miniUI` https://github.com/r-lib/devtools/issues/2388.

Because addins are evaluated in the console (and not a background session<sup>*</sup>), addins should generally be proactive about their dependencies anyway. Most addins are shipped as functions in packages and those packages should list their dependencies as per usual package requirements. 

Furthermore, `rlang::check_installed()` is an excellent and user-friendly way for addin authors to check for installed packages. It helpfully will offer to install missing packages when run interactively.

<sup>*</sup> In other cases where RStudio prompts the user to install an updated package, like rendering an R Markdown document, the code is evaluated in a background session. In these cases, the package is strictly required **and** the if not installed the failure can be hard to work with because of the background session. Neither are true in the case of `miniUI` and addins.
